src/lib/draw/Button.svelte

```diff
@@ -0,0 +1,57 @@
+<script lang="ts">
+  import type { Mode } from "./types";
+
+  export let mode: Mode;
+  export let thisMode: Mode;
+  export let label: string;
+  export let icon: string;
+  export let changeMode: (m: Mode) => void;
+</script>
+
+<button
+  type="button"
+  on:click={() => changeMode(thisMode)}
+  disabled={mode == thisMode}><img src={icon} alt={label} /> {label}</button
+>
+
+<style>
+  button {
+    background-color: #4caf50;
+    border: solid 2px #4caf50;
+    color: white;
+    padding: 15px 32px;
+    text-align: center;
+    text-decoration: none;
+    display: inline-block;
+    font-size: 16px;
+    margin: 4px 2px;
+    cursor: pointer;
+    width: 100%;
+  }
+
+  button:hover:enabled {
+    background-color: white;
+    color: black;
+    transition-duration: 0.4s;
+    border: solid 2px black;
+  }
+
+  button:disabled {
+    opacity: 0.6;
+    cursor: auto;
+  }
+
+  /* The SVG icons are defined with black. By default, make them white, using
```

Comment:
Alternatively, we could use `<svg>` and use simpler CSS to change color. But then we have to figure out the build system... https://natclark.com/tutorials/svelte-working-with-svgs/ and https://joshuatz.com/posts/2021/using-svg-files-in-svelte/#directly-inlining may be relevant. This way works for now, though